libsplinter/src/transport/matrix.rs

```diff
@@ -72,11 +72,18 @@ impl ConnectionMatrixEnvelope {
     }
 
     /// Returns the bytes of the payload while consuming the `ConnectionMatrixEnvelope`
+    #[deprecated(since = "0.3.19", note = "Please use Vec::from instead")]
     pub fn take_payload(self) -> Vec<u8> {
```

Comment:
Interestingly enough, I am no more inclined to think this method should still exist but be called `into_inner`, which is also common for this style of transformation in the standard library.